packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -153,9 +153,34 @@ async function isYarnModulesInstalled(workspacePath: string): Promise<boolean> {
   }
 }
 
-async function isPnpmModulesInstalled(): Promise<boolean> {
-  // TODO: add pnpm support
-  return false;
+async function isPnpmModulesInstalled(workspacePath: string): Promise<boolean> {
+  try {
+    const { stdout } = await command("pnpm ls --json", {
+      cwd: workspacePath,
+      quietErrorsOnExit: true,
+    });
+    const packages = JSON.parse(stdout);
+
+    if (packages && packages.length === 0) {
+      return false;
+    }
+
+    for (const pkg of packages) {
+      if (!pkg || !pkg.dependencies || Object.keys(pkg.dependencies).length === 0) {
+        return false;
+      }
+
+      for (const depInfo of Object.values<{ version: string }>(pkg.dependencies)) {
+        if (!depInfo || !depInfo.version) {
+          return false;
+        }
+      }
+    }
```

Comment:
can you add a few lines of explanation in comments on what do we test here and why?